tests/NuGet.Jobs.Catalog2Registration.Tests/IntegrationTests.cs

```diff
@@ -0,0 +1,451 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Options;
+using Moq;
+using NuGet.Packaging.Core;
+using NuGet.Protocol;
+using NuGet.Protocol.Catalog;
+using NuGet.Services;
+using NuGet.Services.Metadata.Catalog;
+using NuGet.Versioning;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace NuGet.Jobs.Catalog2Registration
+{
+    public class IntegrationTests
+    {
+        [Fact]
+        public async Task AddFirstVersionWhenInlined()
+        {
+            // Arrange & Act
+            await AddVersionsAsync("7.1.2-ALPHA");
+
+            // Assert
+            AssertContainerExistence();
+            AssertBlobExistence(
+                existing: new[]
+                {
+                    "windowsazure.storage/7.1.2-alpha.json",
+                    "windowsazure.storage/index.json",
+                },
+                deleted: new string[0]);
+        }
+
+        [Fact]
+        public async Task AddSecondVersionWhenInlined()
+        {
+            // Arrange
+            await AddVersionsAsync("7.1.2-ALPHA");
+
+            // Act
+            await AddVersionsAsync("7.2.0");
+
+            // Assert
+            AssertContainerExistence();
+            AssertBlobExistence(
+                existing: new[]
+                {
+                    "windowsazure.storage/7.1.2-alpha.json",
+                    "windowsazure.storage/7.2.0.json",
+                    "windowsazure.storage/index.json",
+                },
+                deleted: new string[0]);
+        }
+
+        [Fact]
+        public async Task UpdateExistingVersion()
+        {
+            // Arrange
+            var version = "7.1.2-alpha";
+            await AddVersionsAsync(version);
+
+            // Act
+            await AddVersionsAsync(version);
+
+            // Assert
+            AssertContainerExistence();
+            AssertBlobExistence(
+                existing: new[]
+                {
+                    "windowsazure.storage/7.1.2-alpha.json",
+                    "windowsazure.storage/index.json",
+                },
+                deleted: new string[0]);
+        }
+
+        [Fact]
+        public async Task DeleteWhenNoVersionExists()
+        {
+            // Arrange & Act
+            await DeleteVersionsAsync("7.1.2-ALPHA");
+
+            // Assert
+            AssertContainerExistence();
+            AssertBlobExistence(
+                existing: new string[0],
+                deleted: new[]
```

Comment:
Super nitpick: this name is a tad misleading in this context. Is there another name that could mean "deleted or does not exist"? Maybe `missing`? I don't feel strongly about this, so feel free to keep as is.

Reply:
Yup, sounds fine.